### PR TITLE
Many improvments

### PR DESCRIPTION
**Pull request by AsterAI**
Syntax recognition fixed
- `public static / public final`
- `<SomeType[]>`
- `let this->filter[] = filters;`
- `this->filter["for" : "1"]`
- `unset this->_data[offset];`
- `protected _value { get, __toString };` / `{ get };` / `{ get, set };`
- `return;`
- 
```zep
  let this->_validators = array_filter(validators, function(var element) {
    return typeof element[0] != "array" || !(element[1] instanceof CombinedFieldsValidator);
  });
```

Added keywords highlight: `true`, `false`, `null`

`ZephirParserUtils` have shifted to src scope, to help beginner start easier

Completion list now is shows arguments of method at top of the list, 
also list shows variants doe to context (class, file, method)

### CODE REVIEW

**Comment by AsterAI** [comment]
Need decide about version policy, i started version 0.3* cause to me it's new epoch )))

**Comment by AsterAI** [comment]
[Zephir.zip](https://github.com/phalcon/idea-plugin/files/994431/Zephir.zip)

File: resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <idea-plugin url="https://github.com/zephir-lang/idea-plugin">
   <id>com.zephir</id>
   <name>Zephir</name>
-  <version>0.2.5</version>
+  <version>0.3</version>
```
Comment (sergeyklay) [inline resources/META-INF/plugin.xml]: Could you please use 0.3.0 here?

File: resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -12,6 +12,7 @@
     ]]></description>
 
   <change-notes><![CDATA[
+        <li><b>0.3</b>:  Fixed many bugs with syntax recognition, added few words to highlight, improved completion</li>
```
Comment (sergeyklay) [inline resources/META-INF/plugin.xml]: The same

File: src/com/zephir/completion/providers/FileScopeKeywordsProvider.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import com.intellij.codeInsight.completion.CompletionProvider;
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.psi.PsiElement;
+import com.intellij.util.ProcessingContext;
+import com.zephir.psi.ZephirFile;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Created by alexandr on 09.05.17.
```
Comment (sergeyklay) [inline src/com/zephir/completion/providers/FileScopeKeywordsProvider.java]: Is this comment line needed here?

File: src/com/zephir/completion/providers/FileScopeKeywordsProvider.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ */
+public class FileScopeKeywordsProvider extends CompletionProvider<CompletionParameters> {
+    private String[] keywords = new String[]{
+        "namespace", "use", "as", "interface", "class", "extends", "implements", "final", "abstract"
```
Comment (sergeyklay) [inline src/com/zephir/completion/providers/FileScopeKeywordsProvider.java]: Could you add `internal` and `inline`

**Comment by AsterAI** [comment]
[Zephir.zip](https://github.com/phalcon/idea-plugin/files/997627/Zephir.zip)
Last build - 12 may 2017

**Comment by sergeyklay** [comment]
Thank you

**Pull request merged**